lib/app/app.rb

```diff
@@ -84,51 +85,92 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   post '/status' do
-    source_status = @connector.source_status(body_params)
+    connector = settings.connector_class.new
+
+    source_status = connector.source_status(body_params)
     json(
-      :extractor => { :name => @connector.name },
+      :extractor => { :name => connector.name },
       :contentProvider => source_status
     )
   end
 
+  post '/start_sync' do
+    job = settings.job_store.create_job
+
+    settings.job_runner.start_job(
```

Comment:
Should catch `RejectedExecutionError` here? If the pool is fully utilized and the job is rejected, we should return some meaningful message here?

Reply:
I'm actually not sure about it - our HTTP interface is gonna be frozen in current state and we won't need to bother much about it. Currently it will mark the job as failed with INTERNAL_SERVER_ERROR and the error can be seen in the connector - probably that's sufficient?

Reply:
Based on my understanding, the pool doesn't have a task size limit. I think it will happily queue tasks and never reject a new one.